src/NuGet.Clients/NuGetConsole.Host.PowerShell/PowerShellHost.cs

```diff
@@ -612,10 +602,8 @@ private void RestoreEvents_SolutionRestoreCompleted(SolutionRestoredEventArgs ar
 
         private bool ShouldNoOpDueToRestore(SolutionRestoredEventArgs latestRestore)
         {
-            return latestRestore != null &&
-                   _currentRestore != null &&
-                   latestRestore.SolutionSpecHash == _currentRestore.SolutionSpecHash &&
-                   latestRestore.IsSuccess == _currentRestore.IsSuccess;
+            return _currentRestore != null &&
+                   latestRestore?.RestoreStatus == NuGetOperationStatus.Succeeded;
```

Comment:
Before it did by comparing hashes, now it relies on restore result to be anything but NoOp.